libs/extensions/include/NanoMuon.hpp

```diff
@@ -36,6 +41,8 @@ class NanoMuon {
   MuonID GetID();
   MuonIso GetIso();
 
+  float OuterDeltaRtoMuon(NanoMuon muon);
```

Comment:
This should probably take std::shared_ptr<NanoMuon> as an argument, otherwise, you're making a copy here, no?